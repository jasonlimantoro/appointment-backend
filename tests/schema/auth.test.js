import gql from 'graphql-tag';
import jwt from 'jsonwebtoken';
import { createTestClientAndServer } from '../utils';
import mockUser from '../../fixtures/users';

jest.mock('jsonwebtoken');

describe('Authentication', () => {
  it('login: should work', async () => {
    const decoded = {
      sub: 'uuid-user',
    };
    const { mutate, authAPI, sessionAPI } = createTestClientAndServer();
    authAPI.login = jest.fn().mockResolvedValue('some-random-token');
    jwt.decode = jest.fn().mockReturnValue(decoded);
    sessionAPI.create = jest.fn();
    const LOGIN = gql`
      mutation Login($username: String!, $password: String!) {
        login(username: $username, password: $password)
      }
    `;
    const res = await mutate({ mutation: LOGIN, variables: mockUser });
    expect(res).toMatchSnapshot();
    expect(authAPI.login).toBeCalledWith(mockUser);
    expect(jwt.decode).toBeCalledWith('some-random-token');
    expect(sessionAPI.create).toBeCalledWith({ userId: decoded.sub });
  });
});
