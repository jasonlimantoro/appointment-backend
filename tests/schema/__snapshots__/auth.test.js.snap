// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication login: should work 1`] = `
Object {
  "data": Object {
    "login": Object {
      "session": Object {
        "id": "c29tZS11bmlxdWUtaWQ=",
      },
      "token": "some-token",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Authentication logout: should work 1`] = `
Object {
  "data": Object {
    "logout": true,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
